relayer/src/supervisor/scan.rs

```diff
@@ -815,3 +854,67 @@ fn query_connection_channels<Chain: ChainHandle>(
         })
         .map_err(Error::query)
 }
+
+/// Telemetry discovery is only done for metrics which will be recorded.
+/// For example if the client workers or client misbehaviour detection have
+/// been disabled in the configuration, the `client_misbehaviours_submitted`
+/// will never record any value as no misbehaviour will be submitted.
+fn init_telemetry(
```

Comment:
I think we can clean this function up a bit from a logic perspective by rearranging how the checks are done. Something like this:
```rust
fn init_telemetry(
    chain_id: &ChainId,
    client: &ClientId,
    counterparty_chain_id: &ChainId,
    channel_id: &ChannelId,
    port_id: &PortId,
    config: &Config,
) {
    // Boolean flag that is toggled if any of the tx workers are enabled
    let mut tx_worker_enabled = false;

    let clear_packets = config.mode.packets.enabled
        && (config.mode.packets.clear_on_start || config.mode.packets.clear_interval > 0);

    if config.mode.packets.enabled {
        tx_worker_enabled = true;

        telemetry!(
            init_per_path,
            chain_id,
            counterparty_chain_id,
            channel_id,
            port_id,
            clear_packets
        );

        telemetry!(init_worker_by_type, WorkerType::Packet);

        if config.mode.packets.tx_confirmation {
            telemetry!(init_per_channel, chain_id, channel_id, port_id);
        }
    }

    if config.mode.clients.enabled {
        tx_worker_enabled = true;
        telemetry!(init_worker_by_type, WorkerType::Client);
    }

    if config.mode.connections.enabled {
        tx_worker_enabled = true;
        telemetry!(init_worker_by_type, WorkerType::Connection);
    }

    if config.mode.channels.enabled {
        tx_worker_enabled = true;
        telemetry!(init_worker_by_type, WorkerType::Channel);
    }

    // Now we can just check this boolean
    if tx_worker_enabled {
        telemetry!(
            init_per_client,
            chain_id,
            counterparty_chain_id,
            client,
            config.mode.clients.misbehaviour && config.mode.clients.enabled
        );
    }

    telemetry!(init_worker_by_type, WorkerType::Wallet);
}
```
This removes the `if config.mode.clients.enabled || config.mode.channels.enabled || config.mode.connections.enabled || config.mode.packets.enabled` check so there's a bit less redundancy. 

Of course, if it turns out the metrics need to be initialized in a particular such that this refactor doesn't work, then disregard 🙂

Reply:
This is great!! From what I checked it works, I will test it and if I don't encounter any issue I will push the changes